subsys/net/lib/icalendar_parser/src/icalendar_parser.c

```diff
@@ -0,0 +1,427 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdio.h>
+#include <strings.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <toolchain/common.h>
+#include <net/icalendar_parser.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(icalendar_parser, CONFIG_ICAL_PARSER_LOG_LEVEL);
+
+static size_t unfold_contentline(const char *buf, char *prop_buf)
+{
+	size_t unfold_size = 0;
+	const char *sol = buf;
+	char *eol;
+	u16_t total_line_len = 0, single_line_len = 0;
+
+	while (1) {
+		eol = strstr(sol, "\r\n");
+		if (!eol) {
+			return 0;
+		}
+		single_line_len = eol - sol;
+		if ((total_line_len + single_line_len)
+		    <= CONFIG_ICAL_PARSER_MAX_PROPERTY_SIZE) {
+			memcpy(prop_buf +
+			       total_line_len, sol, single_line_len);
+			total_line_len += single_line_len;
+		} else {
+			LOG_DBG("Property value overflow."
+				"Increase CONFIG_ICAL_PARSER_MAX_PROPERTY_SIZE.");
+			return 0;
+		}
+		if (!strncmp(eol, "\r\n ", strlen("\r\n "))) {
+			/* Long content line is split into multiple lines.
+			 * Copy the folded part to unfolded buffer.
+			 */
+			sol = eol + strlen("\r\n ");
+		} else {
+			/* Content line is delimited. Count parsed bytes */
+			unfold_size = eol - buf;
+			break;
+		}
+	}
+	prop_buf[unfold_size] = '\0';
+
+	return unfold_size;
+}
+
+static bool parse_desc_props(const char *buf,
+			     const char *name,
+			     size_t name_size,
+			     char *value,
+			     size_t max_value_len)
+{
+	bool ret;
+	size_t unfold_size;
+	char ical_prop_buf[CONFIG_ICAL_PARSER_MAX_PROPERTY_SIZE + 1];
```

Comment:
That's 1K that is pushed on the stack right here, that's a lot. Do we need 1K on average, and have you considered using a static buffer instead?